File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        cur = root
        stk = []
        n = 0

        while cur or stk:
            while cur:
                stk.append(cur)
                cur = cur.left
            
            # reach here if cur is None
            cur = stk.pop()

            n += 1
            if n == k:
                return cur.val

            cur = cur.right
        
        return -1

File: Data Structures & Algorithms/remove-element/submission-0.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        var size = nums.Length;
        var left = 0;

        for (int right = 0; right < size; right++)
        {
            if (nums[right] != val)
            {
                (nums[left], nums[right]) = (nums[right], nums[left]);
                left += 1;
            }
        }

        return left;
    }
}